components/greasemonkey.js

```diff
@@ -74,7 +74,8 @@ function startup(aService) {
   parentMessageManager.addMessageListener(
       'greasemonkey:scripts-for-uuid',
       aService.getScriptsForUuid.bind(aService));
-  parentMessageManager.addMessageListener(
+  var mm = Services.ppmm ? Services.ppmm : globalMessageManager;
```

Comment:
Have you tested both versions of this?  It took me quite a while (long enough to be embarrassing) that the "parent process" and "global" message managers cannot be used interchangeably.